content/en/real_user_monitoring/browser/advanced_configuration.md

```diff
@@ -18,10 +18,87 @@ further_reading:
       text: 'Tracking custom user actions'
 ---
 
-## Control sensitive RUM data
-If your RUM data contains sensitive information that need redacting, configure the Browser SDK to redact sensitive sequences, or to discard selected RUM events, by using the `beforeSend` callback when you initialize RUM.
+## Enrich and control RUM data
+The RUM SDK automatically captures most RUM events and populates their main attributes. The `beforeSend` callback function gives you access to every event collected by the RUM SDK before they get sent to Datadog. Intercepting the RUM events allows you to:
+* Enrich your RUM events with extra context attributes
+* Modify your RUM events to modify their content, or redact sensitive sequences
+* Discard selected RUM events
 
-This callback function gives you access to every event collected by the RUM SDK before they get sent to Datadog.
+```javascript
+function beforeSend(event, context)
+```
+
+Starting with [version 2.13.0][3], `beforeSend` takes 2 arguments: the `event` generated by the RUM SDK, and the `context` that triggered the creation of the RUM event. Here is the list of potential `context` values:
+
+| RUM event type   | Context                   |
+|------------------|---------------------------|
+| View             | [Location][4]                  |
+| Action           | [Event][5]                     |
+| Resource (XHR)   | [XMLHttpRequest][6]            |
+| Resource (Fetch) | [Request][7] and [Response][8]      |
```

Comment:
we also have the corresponding PerformanceResourceTiming when available

Reply:
Yes, I think this table could be improved with exact properties and types, at least for JS developers who won't have type hints: https://github.com/DataDog/browser-sdk/blob/main/packages/rum-core/src/rawRumEvent.types.ts#L235-L260

Reply:
I tried to balance between exhaustiveness and making it understandable for the average reader. I also can link to the file but then if we change anything the link is broken... I'm going to add `PerformanceResourceTiming` and that will do the trick. What do you think?

Reply:
It sounds ok-ish if we plan on having a more complete guide quickly, otherwise I'd be in favor of having something more exhaustive first and simplify when we will have a dedicated guide.

Reply:
There's just `Error` missing now then right? What's the difference between `RequestInit` and `RequestInput`?